arangod/Aql/QueryWarnings.cpp

```diff
@@ -53,6 +53,10 @@ void QueryWarnings::registerWarning(int code, std::string const& details) {
   registerWarning(code, details.c_str());
 }
 
+void QueryWarnings::registerWarning(int code, std::string_view details) {
+  registerWarning(code, details.data());
```

Comment:
I don't know if `registerWarning` expects the string to be null-terminated but `data` from a string is not necessarily null-terminated.